sormas-backend/src/test/java/de/symeda/sormas/backend/contact/ContactServiceTest.java

```diff
@@ -129,16 +129,23 @@ public void testGetAllRelevantContacts() {
 		PersonDto contactPerson2 = creator.createPerson();
 		ContactDto contact3 = creator.createContact(user.toReference(), contactPerson2.toReference());
 		contact3.setReportDateTime(referenceDate);
-		getContactFacade().saveContact(contact3);
+		updateContact(contact3);
 
 		ContactDto contact4 = creator.createContact(user.toReference(), contactPerson.toReference(), Disease.CSM);
 		contact4.setReportDateTime(referenceDate);
-		getContactFacade().saveContact(contact4);
+		updateContact(contact4);
 
 		contacts = getContactService().getAllRelevantContacts(contactPersonEntity, contact1.getDisease(), referenceDate);
 		assertThat(contacts, empty());
 	}
 
+	// Workaround for timing issues in the tests:
+	private void updateContact(ContactDto contact) {
```

Comment:
Instead of using a workaround this workaround I'd suggest to use the dto returned by the saveContact call instead.
e.g. `contact1 = getContactFacade().saveContact(contact1)`

Reply:
I thought I tried it like that and it didn't work... Will try again..